tests/integration/components/new-course-test.js

```diff
@@ -0,0 +1,177 @@
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function () {
+    const school = this.server.create('school');
+    this.school = await this.owner.lookup('service:store').find('school', school.id);
+  });
+
+  test('it renders', async function (assert) {
+    const thisYear = (new Date()).getFullYear();
+    await render(
+      hbs`<NewCourse @currentSchool={{this.school}} @save={{noop}} @cancel={{noop}} />`
+    );
+    assert.equal(component.years().count, 6);
+    assert.equal(component.years(0).text, 'Select Academic Year');
+    assert.equal(component.years(1).text, thisYear - 2);
+    assert.equal(component.years(2).text, thisYear - 1);
+    assert.equal(component.years(3).text, thisYear);
+    assert.equal(component.years(4).text, thisYear + 1);
+    assert.equal(component.years(5).text, thisYear + 2);
+    assert.ok(component.years(0).selected);
+    assert.notOk(component.years(1).selected);
+    assert.notOk(component.years(2).selected);
+    assert.notOk(component.years(3).selected);
+    assert.notOk(component.years(4).selected);
+    assert.notOk(component.years(5).selected);
+  });
+
+  test('given year is pre-selected', async function (assert) {
+    const thisYear = (new Date()).getFullYear();
+    const academicYear = this.server.create('academicYear', {id: thisYear});
+    const academicYearModel = await this.owner.lookup('service:store').find('academic-year', academicYear.id);
+    this.set('year', academicYearModel);
+    await render(
+      hbs`<NewCourse @currentSchool={{this.school}} @save={{noop}} @cancel={{noop}} @currentYear={{this.year}} />`
+    );
+    assert.notOk(component.years(0).selected);
+    assert.notOk(component.years(1).selected);
+    assert.notOk(component.years(2).selected);
+    assert.ok(component.years(3).selected);
+    assert.notOk(component.years(4).selected);
+    assert.notOk(component.years(5).selected);
+  });
+
+  test('given year is not pre-selected if it falls out of range', async function (assert) {
+    const thisYear = (new Date('1824-09-04')).getFullYear();
+    const academicYear = this.server.create('academicYear', { id: thisYear });
+    const academicYearModel = await this.owner.lookup('service:store').find('academic-year', academicYear.id);
+    this.set('year', academicYearModel);
+    await render(
+      hbs`<NewCourse @currentSchool={{this.school}} @save={{noop}} @cancel={{noop}} @currentYear={{this.year}} />`
+    );
+    assert.ok(component.years(0).selected);
+    assert.notOk(component.years(1).selected);
+    assert.notOk(component.years(2).selected);
+    assert.notOk(component.years(3).selected);
+    assert.notOk(component.years(4).selected);
+    assert.notOk(component.years(5).selected);
+  });
+
+  // currently doesn't work since the injected ilios-config mock service doesn't seem to get picked up [ST 2021/02/12]
+  skip('year options show range if applicable', async function (assert) {
```

Comment:
The best way to inject config is to mock that API call instead of the service like [this](https://github.com/ilios/frontend/blob/6c6c7e5eccb582b400fa92d0c12593c2205d1d26/tests/acceptance/search-test.js#L14)

Reply:
this saved the day! :heavy_check_mark: